File: solutions/elixir/etl/1/lib/etl.ex
defmodule ETL do
  @doc """
  Transforms an old Scrabble score system to a new one.

  ## Examples

    iex> ETL.transform(%{1 => ["A", "E"], 2 => ["D", "G"]})
    %{"a" => 1, "d" => 2, "e" => 1, "g" => 2}
  """
  @spec transform(map) :: map
  def transform(input) do
    Enum.reduce(input, %{}, fn {points, letters}, acc_input ->
      letters
      |> Enum.reduce(%{}, fn l, acc_letter -> Map.put(acc_letter, String.downcase(l), points) end)
      |> Map.merge(acc_input)
    end)
  end
end

File: solutions/go/etl/1/etl.go
package etl

import "strings"

func Transform(in map[int][]string) map[string]int {
	transformed := make(map[string]int)
	for score, letters := range in {
		for _, l := range letters {
			transformed[strings.ToLower(l)] = score
		}
	}
	return transformed
}
